### PR TITLE
docs(readme): add project background and credit Cursor as the origin idea

### PR DESCRIPTION
**Pull request by xiaoquqi**
## Summary
- Add a "Background" section (English + Chinese) explaining the motivation: traditional graphical-orchestration RAG tools (Dify, n8n) demand a lot of expertise and the real bottleneck is upfront document splitting/embedding — expensive to get right and recall accuracy still disappoints.
- Explain the origin idea: observing that Cursor gets consistently accurate answers over a codebase with zero pre-indexing led to applying the same AI coding agent harness approach to RAG instead of the embed-then-retrieve pipeline.
- Credit Cursor alongside Claude Code/Codex everywhere the README already said "the same kind of harness" — it was left out even though it's the actual origin of the idea.
- Also updated the GitHub repo "About" description to include Cursor for consistency.

## Test plan
- [ ] Render both README.md and README.zh-CN.md on GitHub and confirm the new Background section reads well after the intro paragraph

___

### **PR Type**
Documentation

___

### **Description**
- Add Background section explaining project origin

- Credit Cursor as origin idea alongside Claude Code/Codex

- Replace Mermaid flowchart with product diagram image

- Update Chinese README translation consistently

___

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Add background section and credit Cursor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

<ul><li>Added "Background" section describing motivation and Cursor's role<br> <li> Updated references to credit Cursor alongside Claude Code/Codex<br> <li> Replaced Mermaid flowchart with product diagram image <br>(what_is_sourcelens.png)<br> <li> Updated "Why SourceLens" bullet to include Cursor</ul>

</details>

  </td>
  <td><a href="https://github.com/HyperBDR/sourcelens/pull/89/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+15/-10</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>README.zh-CN.md</strong><dd><code>Update Chinese README correspondingly</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.zh-CN.md

<ul><li>Added Chinese "项目背景" section mirroring English background<br> <li> Updated references to credit Cursor alongside Claude Code/Codex<br> <li> Replaced Mermaid flowchart with product diagram image <br>(what_is_sourcelens.png)<br> <li> Updated "为什么选择 SourceLens" bullet to include Cursor</ul>

</details>

  </td>
  <td><a href="https://github.com/HyperBDR/sourcelens/pull/89/files#diff-13f564fa6ce92394f8423bbcea359aef7e9ebc4f79763efc0d791d9ce5b6f987">+15/-10</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by fengren** [comment]
Reviewed current head `a5993c9` against the PR description and `PR_WORKFLOW.md`.

### Findings

**Important — `README.md:20` and `README.zh-CN.md:20`: the new background makes an inaccurate claim about Cursor.**

The text says Cursor performs “no pre-training or pre-indexing at all.” Cursor relies on pretrained models, while the human-authored PR description only motivates the change from an observed zero-upfront-indexing experience. The current wording therefore overstates what was observed and misattributes the source of accuracy. Please remove the pre-training claim and avoid asserting Cursor's internal indexing behavior; describe SourceLens's own no-embedding/no-pre-indexing design instead.

**Important — `README.md:26` and `README.zh-CN.md:26`: existing Skills support is described as future work.**

The new paragraph says Skills and MCP integration are planned, but packaged Skills are already managed, bound, dispatched to LensNode runs, and presented later in the README as an internal use case “today.” Please distinguish the current Skills capability from genuinely planned extensions so the capability status is internally consistent.

No GitHub Issue is linked, so requirement compliance was assessed from the human-authored PR description. `git diff --check` passed; README rendering was not independently exercised.
